### DISCUSSION TITLE
What are the output of detection_predictor?

### DISCUSSION THREAD

**Discussion by yumikim381** [mindee/doctr#1619] [General]
Hello! I was wondering what exactly detection_predictor returns? 

Let's say output  is the return value from DetectionPredictor([npimg]). 
I get that when you do output [0]["words"], you get an array with 4 coordinates of the bounding boxes and score, 
but the coordinates seems to be all starting with "0. .. "  .Like this : 
`[[0.6015625  0.9375     0.70898438 0.97265625 0.58779669]
 [0.484375   0.93847656 0.55566406 0.97265625 0.5817132 ]
 [0.22265625 0.93652344 0.29394531 0.97558594 0.56156856]
 [0.13085938 0.93847656 0.19238281 0.96777344 0.60816777]`
My question is how are the coordinates ordered: i m guessing they are xmin, ymin, xmax, ymax but is the order correct? 
Also why are the coordinates so small?  How can i get the coordinate of bounding boxes in the original image? 
Thank you and have a nice day!

kind regards, 
Yumi

**Comment by felixdittrich92** [comment]
Hi @yumikim381 :wave:,

Correct the format is `[xmin, ymin, xmax, ymax, score]` and the coordinates a relative so you need to multiply the x values with the width and y values with the height of your image :)

**Comment by yumikim381** [comment]
That helps so much! 
Thanks a lot for your swift reply

**Comment by yumikim381** [comment]
One more question, if i give such images of one line to DocTR detection_predictor, i get no detection results.
![image](https://github.com/mindee/doctr/assets/102374351/670a75e7-816a-42fc-bf65-2f65029934ce)
Do you by chance know if this phenomen is due to the lack of words in those images or due to preprocessing(scaling)? 
I know that docTR line detector is not trained to detect on images with single line on it, but doctr's ocr_predictor seem to be able to detect on single line images. 
<img width="674" alt="image" src="https://github.com/mindee/doctr/assets/102374351/6d80fc29-19c1-4080-b509-7e1951b54d41">
Above is the result of doing :
```
result = predictor(doc)
result.show()
synthetic_pages = result.synthesize()
plt.imshow(synthetic_pages[0]); plt.axis('off');
``` 
So i m wondering if this is bug in my code. . Do you know how to fix this?

**Reply by felixdittrich92** [reply]
So in short with the `ocr_predictor` it works but not with the `detection_predictor` ? 
If so the issue is in your code because the `ocr_predictor` is in general only a wrapper around the `detection_predictor` and `recognition_predictor`

Addition: Oh i think i missed to mention that you need to remove the padding from the results :/

https://github.com/mindee/doctr/blob/9c92f5cbd0bf64706672c29bf2d43117815c1794/doctr/models/predictor/base.py#L107

**Reply by felixdittrich92** [reply]
CC @odulcy-mindee @frgfm i think we should move this into the detection predictor call !? wdyt ?

**Reply by yumikim381** [reply]
Ohh is this for removing paddings from detection results before giving it to recognizer? Do you know why it is necessary to do this?

**Reply by felixdittrich92** [reply]
Each page is resized to 1024x1024 (default value which was also used for training) for detection ..additional there are multiple options like symmetric_pad and preserve_aspect_ratio so with remove_padding we keep all these option values into account to transform the results back to the original input page shape :)

**Reply by yumikim381** [reply]
Ahh okay thank you so much for the infos!!

**Reply by felixdittrich92** [reply]
Already opened a PR to fix this: https://github.com/mindee/doctr/pull/1627

**Comment by appsembly-team** [comment]
![image](https://github.com/user-attachments/assets/4937efb5-01d8-46c3-95f2-c650623a041c)
I am getting like this. Can someone help related to this to  xmin, ymin, xmax, ymax and score? Thanks
